app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt

```diff
@@ -0,0 +1,593 @@
+import org.mozilla.telemetry.event.TelemetryEvent
+import org.mozilla.telemetry.measurement.DefaultSearchMeasurement
+import org.mozilla.telemetry.measurement.SearchesMeasurement
+import org.mozilla.telemetry.net.HttpURLConnectionTelemetryClient
+import org.mozilla.telemetry.ping.TelemetryCorePingBuilder
+import org.mozilla.telemetry.ping.TelemetryEventPingBuilder
+import org.mozilla.telemetry.schedule.jobscheduler.JobSchedulerTelemetryScheduler
+import org.mozilla.telemetry.serialize.JSONPingSerializer
+import org.mozilla.telemetry.storage.FileTelemetryStorage
+
+@Suppress(
+        // Yes, this a large class with a lot of functions. But it's very simple and still easy to read.
+        "TooManyFunctions",
+        "LargeClass"
+)
+object TelemetryWrapper {
+    private const val TELEMETRY_APP_NAME_FOCUS = "Focus"
+    private const val TELEMETRY_APP_NAME_KLAR = "Klar"
+
+    private const val MAXIMUM_CUSTOM_TAB_EXTRAS = 10
+
+    private val isEnabledByDefault: Boolean
+        get() = !AppConstants.isKlarBuild()
+
+    private object Category {
+        val ACTION = "action"
+    }
+
+    private object Method {
+        val TYPE_URL = "type_url"
+        val TYPE_QUERY = "type_query"
+        val TYPE_SELECT_QUERY = "select_query"
+        val CLICK = "click"
+        val SWIPE = "swipe"
+        val CANCEL = "cancel"
+        val LONG_PRESS = "long_press"
+        val CHANGE = "change"
+        val FOREGROUND = "foreground"
+        val BACKGROUND = "background"
+        val SHARE = "share"
+        val SAVE = "save"
+        val COPY = "copy"
+        val OPEN = "open"
+        val INSTALL = "install"
+        val INTENT_URL = "intent_url"
+        val INTENT_CUSTOM_TAB = "intent_custom_tab"
+        val TEXT_SELECTION_INTENT = "text_selection_intent"
+        val SHOW = "show"
+        val HIDE = "hide"
+        val SHARE_INTENT = "share_intent"
+        val REMOVE = "remove"
+        val REORDER = "reorder"
+    }
+
+    private object Object {
+        val SEARCH_BAR = "search_bar"
+        val ERASE_BUTTON = "erase_button"
+        val SETTING = "setting"
+        val APP = "app"
+        val MENU = "menu"
+        val BACK_BUTTON = "back_button"
+        val NOTIFICATION = "notification"
+        val NOTIFICATION_ACTION = "notification_action"
+        val SHORTCUT = "shortcut"
+        val BLOCKING_SWITCH = "blocking_switch"
+        val BROWSER = "browser"
+        val BROWSER_CONTEXTMENU = "browser_contextmenu"
+        val CUSTOM_TAB_CLOSE_BUTTON = "custom_tab_close_but"
+        val CUSTOM_TAB_ACTION_BUTTON = "custom_tab_action_bu"
+        val FIRSTRUN = "firstrun"
+        val DOWNLOAD_DIALOG = "download_dialog"
+        val ADD_TO_HOMESCREEN_DIALOG = "add_to_homescreen_dialog"
+        val HOMESCREEN_SHORTCUT = "homescreen_shortcut"
+        val TABS_TRAY = "tabs_tray"
+        val RECENT_APPS = "recent_apps"
+        val APP_ICON = "app_icon"
+        val AUTOCOMPLETE_DOMAIN = "autocomplete_domain"
```

Comment:
Should add this to the telemetry wiki documentation.

Reply:
Yep. I was waiting for the merge before updating the wiki. :)